src/mraa.c

```diff
@@ -1210,7 +1210,7 @@ mraa_init_io(const char* desc)
         return NULL;
     }
 
-    if (strncmp(type, "GPIO", 4) == 0) {
+    if (strncmp(type, GPIO_KEY, 4) == 0) {
```

Comment:
sizeof(GPIO_KEY) should be replace by the compiler with a value.Not sure how well this works.